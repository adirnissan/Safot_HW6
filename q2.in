use_module(library(yall)).
print("===TEST START===").

once(topological_sort(graph([[2, 3], [], [2, 4], [3]]), S)).

{L}/(
    findall(
        S,
        topological_sort(graph([[2, 3], [], [2, 4], []]), S),
        L1
    ),
    msort(L1, L)
).


{L}/(
    once(scc(graph([[2, 3], [], [2, 4], [3]]), S)),
    findall(
        C1,
        (
            member(C, S),
            msort(C, C1)
        ),
        L1
    ),
    msort(L1, L)
).
