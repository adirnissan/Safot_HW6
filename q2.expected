
false.

L = [[1, 3, 2, 4], [1, 3, 4, 2]].

L = [[1], [2], [3, 4]].
